pkg/runtime/runtime.go

```diff
@@ -1295,6 +1295,12 @@ func (a *DaprRuntime) processComponentAndDependents(comp components_v1alpha1.Com
 	}
 
 	compCategory := a.extractComponentCategory(comp)
+	if compCategory == "" {
+		// the category entered is incorrect, return error
+		err := errors.Errorf("incorrect type %s", comp.Spec.Type)
```

Comment:
we do not need to log error in this method (L1301 and L1305). Instead, we log error in caller code and reduce the duplicated code.
the below code is the only one caller part calling `processComponentAndDependents`. We can print out returned error. 

https://github.com/dapr/dapr/blob/cd35c75320423c7396fcd8bc6abfccdcc52427f4/pkg/runtime/runtime.go#L1274-L1279

Reply:
Good catch. Fixed.